### PR TITLE
feat: support htmlwebpackplugin v4

### LINKED ISSUE #30: support html-webpack-plugin@4
```
(node:13342) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'tapPromise' of undefined
    at compiler.hooks.compilation.tap (node_modules/webpack-babel-multi-target-plugin/src/babel.multi.target.html.updater.ts:112:59)
```

### PR DESCRIPTION
**Pull request by Austaras**
fix #30

### CODE REVIEW

**Comment by erezvish** [comment]
Oh wow, I see you got this already.
Thanks for making this fix!

**Comment by erezvish** [comment]
@DanielSchaffer 
Who is supposed to review the changes, one of the current collaborators, you, or anyone interested?

**Comment by DanielSchaffer** [comment]
@erezvish - hi, I’m sorry for the lack of communication! I’m in the process of switching jobs, so I’ve been scrambling to wrap things up and haven’t had time to look. I’ll try to set aside some time today to take a look so you guys can get unblocked.

**Comment by DanielSchaffer** [comment]
@Austaras, @erezvish - my apologies for the delay - there were a _ton_ of things that came up last minute while trying to leave my former coworkers in a reasonably caught up state, and I wasn't able to get to this until now.

What you've got so far looks great, but I must also apologize that I forgot that the `HtmlWebpackPlugin` hooks are also referenced in `NormalizeModuleIdsPlugin` - those will need to use the same logic you added to `BabelMultiTargetHtmlUpdater`. It might make sense to move your `isV4` and `hook` logic into functions their own file.

I'd recommend trying your changes out by using the examples - you can start the dev server by running this from the command line:

```
NODE_OPTIONS="--max-old-space-size=8192" yarn start
```

Unfortunately, the server must be restarted when you make changes to the plugin code - I'm not sure there's even a way to do incremental builds for a Webpack plugin.

There are several examples, so the initial build may take a while to complete, especially on older machines - but once it's done, you can view the example projects by visiting http://localhost:3002/examples/angular/, and then following the links to each of the other examples. If all of the examples turn the background green after loading, you should be good to go.

Additionally, while I realize I haven't set up a formal "contributing guidelines", I'd much appreciate it if you could do the following:
- use `git rebase -i` to squash your commits into logical chunks - multiple commits are fine if they do unrelated things, but I'd prefer to avoid sequential commits that are something to the effect of "do the thing", and then "that didn't work, actually do the thing"
- please follow the guidelines established by [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) for your commit messages.

Thank you so much for your contributions on this, it is much appreciated!

EDIT: I've added [CONTRIBUTING](https://github.com/DanielSchaffer/webpack-babel-multi-target-plugin/blob/develop/CONTRIBUTING.md)

**Comment by Austaras** [comment]
Sorry for the dealy. Change as you requested. The Angular routing seems like broken in your branch too,  other exmaple works fine

**Comment by Austaras** [comment]
@DanielSchaffer would you take some time review my updated commit?

**Comment by DanielSchaffer** [comment]
@Austaras I believe the broken Angular example is just due to using the wrong typescript version .. I'll fix that in another commit

EDIT: There's a [fix](https://github.com/DanielSchaffer/webpack-babel-multi-target-plugin/commit/46552b3df66213e78d1b020edc4fb155663c6c84) on `develop` if you want to merge or rebase on your branch

File: src/html-webpack-plugin.polyfill.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { getHooks, AlterAssetTagsData, HtmlTag } from 'html-webpack-plugin'
+import { compilation } from 'webpack'
+import Compilation = compilation.Compilation
+
+const isV4 = typeof getHooks !== undefined
```
Comment (DanielSchaffer) [inline src/html-webpack-plugin.polyfill.ts]: `undefined` needs to be quoted

```suggestion
const isV4 = typeof getHooks !== 'undefined'
```

**Review by DanielSchaffer — CHANGES_REQUESTED** [review]
The missing quotes on `undefined` is breaking it when used with `html-webpack-plugin@^3` - make sure you run the examples with both versions to ensure it works correctly. If you've got a `dist` directory from building the plugin previously, you may need to run `yarn build` after installing a different version so the change gets picked up.

**Review by DanielSchaffer — APPROVED** [review]

**Comment by DanielSchaffer** [comment]
@Austaras @erezvish - this will be released as `2.5.0` (note: `2.4.0` is skipped due a some tagging/versioning snafu). Unfortunately, I'm having some issues with CI as Travis seems to have moved my cheese a bit - I can't poke at it right now, but I'll try and come back to it later this afternoon.

**Comment by erezvish** [comment]
Thank you both for this, it is much appreciated!
I'll try the new version when released, no worries though if integration takes a little time.
Thank you again

**Comment by DanielSchaffer** [comment]
@Austaras @erezvish pre-release: [`2.5.0-next.1`](https://www.npmjs.com/package/webpack-babel-multi-target-plugin/v/2.5.0-next.1)

Let me know if that works as expected for you and I'll cut the final `2.5.0` release. Thanks for your efforts and for bearing with me and my delays!

**Comment by Austaras** [comment]
@DanielSchaffer Many thanks for your great plugin!  I tried in my project and find few minor problems
1. issue #56 
2. typing for babel-loader is not bundled so `fork-ts-checker` will compain about it

evrything else works fine

**Comment by erezvish** [comment]
Hi, 
I can confirm that two builds are being created with Html webpack plugin (beta 8).
I'll need to work with it more to see if it's working properly.

Thanks for all the effort!

**Comment by DanielSchaffer** [comment]
@Austaras can you write up an issue for this with repro steps?

> typing for babel-loader is not bundled so fork-ts-checker will complain about it

**Pull request merged**